### ISSUE TITLE
Get rid of manager-release branch support in testing framework (SCT, DTest)

### ISSUE THREAD

**Issue by mikliapko** [scylladb/scylla-manager#4043]
Currently, we are supporting two branches in two testing frameworks used for Manager testing (SCT, DTest):
- **master**;
- **manager-release** (manager-3.3 currently). 

Since I started working with Manager test about half a year ago, we have never encountered a case where there were test-specific differences between these two branches. At the same time, **manager-release** branch support takes time on backporting and conflicts resolution.

To sum it up, I see no reason to support both of them and suggest to stay with only one **master** branch.

If there's a need for a separate release branch, we can create it any time.

**Comment by karol-kokoszka** [comment]
We are not really backporting fixes from master to branches other than the current release.
It makes sense to get rid of manager-release branch in SCT and Dtest.

**Comment by mikliapko** [comment]
Starting from release 3.4 we switched to one branch support approach.
Corresponding Jenkins spaces have been updated.
